src/hoc/ModalHOC.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class ModalHOC extends React.Component {
```

Comment:
This class isn't HOC. It's just wrapper component. HOC it's a function which accepts another component as first argument and returns new component.
Also, usual naming for files it's `with...`, so in your case it should be like `withModal` for file and `WithModal` for component.

Reply:
https://reactjs.org/docs/higher-order-components.html

Reply:
https://reactjs.org/docs/render-props.html

Reply:
Fixed.